File: java/app/src/main/java/net/fbvictorhugo/j/barreirasanitaria/data/dao/dummy/PessoaDummyDataBase.java
package net.fbvictorhugo.j.barreirasanitaria.data.dao.dummy;

import net.fbvictorhugo.j.barreirasanitaria.data.dao.IPessoaDAO;
import net.fbvictorhugo.j.barreirasanitaria.data.model.Pessoa;

import java.util.ArrayList;
import java.util.List;

public class PessoaDummyDataBase implements IPessoaDAO {

    private static PessoaDummyDataBase sFactory;
    private List<Pessoa> mDummyList;

    public static PessoaDummyDataBase getInstance() {
        if (sFactory == null) {
            sFactory = new PessoaDummyDataBase();
        }
        return sFactory;
    }

    private PessoaDummyDataBase() {
        mDummyList = new ArrayList<>();
    }

    public List<Pessoa> listar() {
        return mDummyList;
    }

    @Override
    public void inserir(Pessoa pessoa) throws Exception {
        //TODO("Not yet implemented")
    }

    @Override
    public void atualizar(Pessoa pessoa) throws Exception {
        //TODO("Not yet implemented")
    }

    private long getProximoIdDisponivel() {
        if (mDummyList.size() == 0) {
            return 1;
        } else {
            return mDummyList.get(mDummyList.size() - 1).getId() + 1;
        }
    }

}

File: kotlin/app/src/main/java/net/fbvictorhugo/k/barreirasanitaria/data/dao/dummy/PessoaDummyDataBase.kt
package net.fbvictorhugo.k.barreirasanitaria.data.dao.dummy

import net.fbvictorhugo.k.barreirasanitaria.data.dao.IPessoaDAO
import net.fbvictorhugo.k.barreirasanitaria.data.model.Pessoa

object PessoaDummyDataBase : IPessoaDAO {

    private val mDummyList: ArrayList<Pessoa> = ArrayList()

    init {
        val pessoa = Pessoa("João das Neves")
            .apply {
                id = 1
                numeroDocumento = 10020010052
            }
        mDummyList.add(pessoa)

        val pessoa2 = Pessoa("Fulano de Tal")
            .apply {
                id = 2
                numeroDocumento = 10020010042
            }
        mDummyList.add(pessoa2)
    }

    override fun listar(): ArrayList<Pessoa> {
        return mDummyList
    }

    override fun inserir(pessoa: Pessoa) {
        TODO("Not yet implemented")
    }

    override fun atualizar(pessoa: Pessoa) {
        TODO("Not yet implemented")
    }

}
